File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;
class Solution {
    public int solution(int[] order) {
        int answer = 0;
        
        Stack <Integer> stack = new Stack<>();
        int check= 1;
        
        for (int target : order) {
            if ((!stack.isEmpty()) && (stack.peek() == target)) {
                    answer+=1;
                    stack.pop();
                    continue;
                }
        
  
                for (int i= check ; i <=target; i ++) {
                    stack.push(i);
                }
            
            check = target+ 1;
            
                
            
            if ((!stack.isEmpty() )&& stack.peek() == target) {
                answer +=1;
                stack.pop();
            }
            else break;
            
            
        }
        
        return answer;
            
        }
}

File: 프로그래머스/2/131704. 택배상자/택배상자.py
from collections import deque
def solution(order):
      
    answer = 0
    main_container = [i for i in range (1, len(order)+1)]
    temp_container = []
    check_idx = 0
    for target in order :
        if target not in temp_container :
            for i in range (check_idx, len(main_container)):
                temp_container.append(main_container[i])
                if target == main_container[i] :
                    check_idx = i+1
                    break
        

        if temp_container and temp_container[-1] == target :
            answer +=1
            temp_container.pop(-1)
        
        else:
            break
            
        
        
    return answer